src/App.tsx

```diff
@@ -15,49 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  az = 'az',
+  length = 'length',
+  reverse = 'reverse',
+  default = '',
+}
+
+function getSortedGoods(goods: string[], type: SortType[]) {
+  const sortedGoods = [...goods];
+
+  if (type[0] === SortType.az) {
+    sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (type[0] === SortType.length) {
+    sortedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (type[1] === SortType.reverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType]
+  = React.useState<SortType[]>([SortType.default, SortType.default]);
```

Comment:
```suggestion
  const [sortType, setSortType] = useState<SortType[]>(
    [SortType.default, SortType.default]
  );
```
I don't think that it's a good idea to set array of types. Just create a separate state for reverse